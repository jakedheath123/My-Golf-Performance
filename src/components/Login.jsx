import React from "react"
import TextField from '@material-ui/core/TextField';
import Grid from '@material-ui/core/Grid';
import PersonIcon from '@material-ui/icons/Person';
import LockIcon from '@material-ui/icons/Lock';
import Paper from '@material-ui/core/Paper';


const Login = () => {

  return (
    <div className="root">
      <Paper className="container">
      <Grid container spacing={1} alignItems="flex-end" justify="center" >
          <Grid item >
            <PersonIcon />
          </Grid>
          <Grid item>
            <TextField id="input-with-icon-grid" label="Username" />
          </Grid>
        </Grid>
        <Grid container spacing={1} alignItems="flex-end" justify="center">
          <Grid item>
            <LockIcon />
          </Grid>
          <Grid item>
            <TextField id="input-with-icon-grid" label="Password" />
          </Grid>
        </Grid>
        </Paper>
    </div>
  );
};

export default Login;