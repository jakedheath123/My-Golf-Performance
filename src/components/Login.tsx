import React, { FunctionComponent } from "react"
import { RouteComponentProps } from "@reach/router"
import TextField from '@material-ui/core/TextField';
import Grid from '@material-ui/core/Grid';
import PersonIcon from '@material-ui/icons/Person';
import LockIcon from '@material-ui/icons/Lock';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import ExitToAppIcon from '@material-ui/icons/ExitToApp';
import Typography from '@material-ui/core/Typography';
import FacebookIcon from '@material-ui/icons/Facebook';

const Login: FunctionComponent<RouteComponentProps> = () => { 
  return (
    <div className="login" data-test="component-login">
      <Paper className="login__form">
      <form >
      <Grid container justify="center">
          <Grid item className="login__icon-buttons--spacing">
        <Button 
        className="login__icon-buttons"
        disableElevation
        variant="contained"
        color="secondary"
        startIcon={<i className="fab fa-google"></i>}
        >
       Sign in with Google
      </Button>
      </Grid>  
      </Grid>
      <Grid container justify="center">
          <Grid item className="login__icon-buttons--spacing">
        <Button 
        className="login__icon-buttons"
        disableElevation
        startIcon={<FacebookIcon />}
        variant="contained"
        color="primary">
      Sign in with Facebook
      </Button>
      </Grid>  
      </Grid>
      <Grid container justify="center">
            <Grid item>
            <Typography variant="subtitle1">Or</Typography>
          </Grid>
      </Grid>      
      <Grid container spacing={1} alignItems="flex-end" justify="center" >
          <Grid item >
            <PersonIcon />
          </Grid>
          <Grid item>
            <TextField id="input-with-icon-grid" label="Username" />
          </Grid>
        </Grid>
        <Grid container spacing={1} alignItems="flex-end" justify="center">
          <Grid item>
            <LockIcon />
          </Grid>
          <Grid item>
            <TextField id="input-with-icon-grid" label="Password" />
          </Grid>
          <Grid container justify="center">
            <Grid item className="login__icon-buttons--spacing">
            <Button
            disableElevation
            size="medium"
        variant="contained"
        color="default"
        endIcon={<ExitToAppIcon />}
      >
        Login
      </Button>
            </Grid>
          </Grid>
        </Grid>
        </form>
        </Paper>
    </div>
  );
};

export default Login;