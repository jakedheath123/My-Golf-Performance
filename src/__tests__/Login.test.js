import React from "react";
import { shallow } from "enzyme";
import { create } from "react-test-renderer";

import Login from "../components/Login";

const setup = (props = {}) => shallow(<Login {...props} />);

const findByTestAttr = (wrapper, value) =>
  wrapper.find(`[data-test="${value}"]`);

describe("Login component test suite", () => {
  const wrapper = setup();

  test("matches snapshot", () => {
    const loginSnapshot = create(<Login />).toJSON();
    expect(loginSnapshot).toMatchSnapshot();
  });
  test("renders without an error", () => {
    const loginComponent = findByTestAttr(wrapper, "component-login");
    expect(loginComponent.length).toBe(1);
  });
});
