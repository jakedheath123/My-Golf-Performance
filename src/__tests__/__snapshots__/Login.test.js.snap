// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component test suite matches snapshot 1`] = `
<div
  className="login"
  data-test="component-login"
>
  <div
    className="MuiPaper-root login__form MuiPaper-elevation1 MuiPaper-rounded"
  >
    <form>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root login__icon-buttons--spacing MuiGrid-item"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained login__icon-buttons MuiButton-containedSecondary MuiButton-disableElevation"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <span
                className="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <i
                  className="fab fa-google"
                />
              </span>
              Sign in with Google
            </span>
          </button>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root login__icon-buttons--spacing MuiGrid-item"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained login__icon-buttons MuiButton-containedPrimary MuiButton-disableElevation"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <span
                className="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
                  />
                </svg>
              </span>
              Sign in with Facebook
            </span>
          </button>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-subtitle1"
          >
            Or
          </h6>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-flex-end MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              htmlFor="input-with-icon-grid"
              id="input-with-icon-grid-label"
            >
              Username
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="input-with-icon-grid"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-flex-end MuiGrid-justify-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
            />
          </svg>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink={false}
              htmlFor="input-with-icon-grid"
              id="input-with-icon-grid-label"
            >
              Password
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="input-with-icon-grid"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
        >
          <div
            className="MuiGrid-root login__icon-buttons--spacing MuiGrid-item"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-disableElevation"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Login
                <span
                  className="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
